### PR TITLE
First attempt at a "make test:all" run.

### PR DESCRIPTION
**Pull request by tzarc**
## Description

Adds a new workflow to run `make test:all` when core changes are detected.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: .github/workflows/make_test.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+jobs:
+  test:
+    name: 'make test:all'
+    runs-on: ubuntu-latest
+    container: qmkfm/base_container
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```
Comment (zvecr) [inline .github/workflows/make_test.yaml:25]:
```suggestion
        fetch-depth: 0
        submodules: recursive
```

File: .github/workflows/make_test.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        git rev-parse --short HEAD
+        echo ${{ github.event.pull_request.base.sha }}
+        echo '${{ steps.file_changes.outputs.files}}'
+
+    - name: Install dependencies
+      run: |
+        make git-submodule
+        python3 -m pip install -U qmk
+        python3 -m pip install -r requirements.txt
+
```
Comment (zvecr) [inline .github/workflows/make_test.yaml:43]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: .github/workflows/make_test.yaml (hunk 2 above)
Comment (tzarc) [reply .github/workflows/make_test.yaml:43]: I'd still prefer to keep the updating of `requirements.txt` -- core changes that add anything new will hang during execution, as it asks whether or not pip should be executed during build.

Comment (zvecr) [reply .github/workflows/make_test.yaml:43]: Base container is meant to contain all the latest requirements, be it that its mostly manual right now. But i guess the CI for CLI does it.

```suggestion
    - name: Install dependencies
      run: pip3 install -r requirements.txt
```

**Pull request closed without merging**